protocol/storage/client.go

```diff
@@ -169,14 +170,30 @@ func (smc *Client) ProposeDeal(ctx context.Context, miner address.Address, data
 	proposal.Payment.ChannelMsgCid = &cpResp.ChannelMsgCid
 	proposal.Payment.Vouchers = cpResp.Vouchers
 
+	// client signs the DealProposal with the address they are sending from
```

Comment:
Consider breaking this out to a well-named function, removing the need for this comment.